test/http-api.test.ts

```diff
@@ -247,6 +247,26 @@ describe('HTTP API', () => {
       );
       assert.ok(retryCounter === 0);
     });
+
+    it('throws after 5 seconds timeout', async () => {
+      nock(loginUrl)
+        .get('/services/data/v59.0')
+        .delay(7000) // Delay response by 7 seconds to exceed timeout
+        .reply(200, { success: true });
+
+      const { err } = await fetch(
+        {
+          method: 'GET',
+          url: `${loginUrl}/services/data/v59.0`,
+        },
+        {
+          timeout: 5000, // 5 second timeout
+        },
+      );
```

Comment:
this test validates that `transport.httpRequest`(wrapped in `fetch` here) still allows to pass a custom timeout.